tmd/areas/create_area_weights.py

```diff
@@ -151,15 +145,38 @@ def loss_function_value(wght, target_matrix, target_array):
     return 0.5 * np.sum(np.square(act_minus_exp))
 
 
+def objective_function(x, *args):
+    """
+    Objective function for minimization of sum of squared deviations.
+    """
+    A, b, lambda_ = args
+    target_deviations = A @ x - b  # JAX sparse matrix-vector multiplication
+    weight_deviations = jnp.sqrt(lambda_) * (x - 1)
+    deviations = jnp.concatenate([target_deviations, weight_deviations])
+    return jnp.sum(jnp.square(deviations))
+
+
+JIT_FVAL_AND_GRAD = jax.jit(jax.value_and_grad(objective_function))
```

Comment:
I love the speed-up. This is 30% faster on my machine on the zz problem, which is our hardest problem. Seems like this will really help when we need a lot of iterations. And a good learning experience for me - I am not terribly familiar with python, I have not used jax.jit until this exercise. Thanks.

Reply:
Yes, the jax library is very nice.